lib/Parse/ParseDecl.cpp

```diff
@@ -3332,6 +3332,8 @@ bool Parser::parseDeclList(SourceLoc LBLoc, SourceLoc &RBLoc,
 }
 
 bool Parser::canDelayBodyParsing() {
+  if (IsParsingInterfaceTokens)
+    return false;
```

Comment:
I think this disables "delay parsing" effectively.
`llvm::SaveAndRestore<bool> S(P.IsParsingInterfaceTokens, true)` is at `swift::parseIntoSourceFile()` (it's in `ParseSIL.cpp` but this is the main entry point of parsing)

Reply:
ah, does it mean we always collect interface tokens for hashing and this requires no delayed parsing at all?

Reply:
Yeah, something like this:
```swift
/*<enabled>*/
class C {
  var value = 12 // This affects interface-hash because modifying the initializer might change the type of the property. 

  func foo<T>()  where T : P /*<disabled>*/{
    // Here, no token affects interface-hash.
  }/*</disabled>*/

  func bar() -> Baz /*<disabled>*/{
  }/*</disabled>*/
}
/*</enabled>*/

Reply:
thanks for the example! probably the only parsing can be delayed is for function bodies then :(